tkupchyn/homework_09/fraction.py

```diff
@@ -0,0 +1,44 @@
+from math import gcd
+
+
+class Fraction:
+
+    def __init__(self, nominator: (int, str, float), denominator: int = 1):
+
+        if isinstance(nominator, float):
+            nominator = str(nominator)
+
+        if isinstance(nominator, str):
+            if '.' in nominator:
+                integer_part, fractional_part = nominator.split('.')
+                nominator = int(integer_part + fractional_part)
+                denominator = 10 ** len(fractional_part)
+
+            elif '/' in nominator:
+                nominator, denominator = nominator.split('/')
+                nominator = int(nominator)
+                denominator = int(denominator)
```

Comment:
I see you override denominator, but what if it is specified?

Fraction(0.5,  2)       ->       Fraction(1, 4)